lib/ruby_lsp/server.rb

```diff
@@ -172,6 +177,14 @@ module RubyLsp
       end
     end
 
+    on("textDocument/inlayHint", parallel: true) do |request|
+      document = store.get(request.dig(:params, :textDocument, :uri))
+
+      if document.parsed?
+        Requests::InlayHints.new(document).run
+      end
```

Comment:
Feedback on this? Is this the right thing to put here?

Reply:
The `document.parsed?` aspect is indeed correct, but I think we're missing the range parameter. The [spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_inlayHint) mentions that inlay hints are computed for the current visible range.

Reply:
Right, OK! I'll add that, my bad.